File: frontend/src/features/recruiter/types/recruiterView.ts
export interface RecruiterSignals {
  readability: number;

  impact_density: number;

  technical_depth: number;

  visual_hierarchy: number;
}

export interface RecruiterFeedbackCard {
  title: string;

  description: string;

  confidence: string;

  type: string;
}

export interface RecruiterAttentionZone {
  section: string;
  label: string;
  x: number;
  y: number;
  width: number;
  height: number;
  attention: number;
  start_second: number;
  end_second: number;
  reason: string;
  priority: "high" | "medium" | "low" | string;
}

export interface RecruiterScanPathPoint {
  section: string;
  x: number;
  y: number;
  second: number;
  label: string;
}

export interface RecruiterDropOffPoint {
  second: number;
  section: string;
  reason: string;
  severity: "low" | "medium" | "high" | string;
}

export interface RecruiterTimelineEvent {
  second: number;
  title: string;
  description: string;
  sentiment: "positive" | "neutral" | "negative" | string;
}

export interface RecruiterViewResponse {
  recruiter_score: number;

  signals: RecruiterSignals;

  strengths: string[];

  weak_spots: string[];

  missing_impact: string[];

  ai_feedback: RecruiterFeedbackCard[];

  attention_zones?: RecruiterAttentionZone[];

  scan_path?: RecruiterScanPathPoint[];

  drop_off_points?: RecruiterDropOffPoint[];

  recruiter_timeline?: RecruiterTimelineEvent[];
}

export interface RecruiterViewRequest {
  language: string;

  resume_content: string;

  target_role?: string;

  resume_id?: number | string;
}

export interface SavedRecruiterViewResponse {
  id: number;

  user_id: number;

  resume_id: number;

  recruiter_score: number | null;

  analysis: RecruiterViewResponse | null;

  analyzed_at: string;
}

File: backend/app/schemas/recruiter/recruiter_view.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class RecruiterSignal(BaseModel):
    readability: int
    impact_density: int
    technical_depth: int
    visual_hierarchy: int


class RecruiterFeedbackCard(BaseModel):
    title: str
    description: str
    confidence: str
    type: str

class RecruiterAttentionZone(BaseModel):
    section: str
    label: str
    x: int
    y: int
    width: int
    height: int
    attention: int
    start_second: float
    end_second: float
    reason: str
    priority: str


class RecruiterScanPathPoint(BaseModel):
    section: str
    x: int
    y: int
    second: float
    label: str


class RecruiterDropOffPoint(BaseModel):
    second: float
    section: str
    reason: str
    severity: str


class RecruiterTimelineEvent(BaseModel):
    second: float
    title: str
    description: str
    sentiment: str


class RecruiterViewRequest(BaseModel):
    language: str
    resume_content: str
    target_role: str | None = None
    resume_id: int | None = None


class RecruiterViewResponse(BaseModel):
    recruiter_score: int
    signals: RecruiterSignal
    strengths: List[str]
    weak_spots: List[str]
    missing_impact: List[str]
    ai_feedback: List[RecruiterFeedbackCard]
    attention_zones: List[RecruiterAttentionZone] = []
    scan_path: List[RecruiterScanPathPoint] = []
    drop_off_points: List[RecruiterDropOffPoint] = []
    recruiter_timeline: List[RecruiterTimelineEvent] = []

class SavedRecruiterViewResponse(BaseModel):
    id: int
    user_id: int
    resume_id: int
    recruiter_score: Optional[int] = None
    analysis: Optional[RecruiterViewResponse] = None
    analyzed_at: datetime

    class Config:
        from_attributes = True
